engine/src/org/pentaho/di/trans/steps/datagrid/DataGrid.java

```diff
@@ -84,16 +84,19 @@ public boolean processRow( StepMetaInterface smi, StepDataInterface sdi ) throws
     List<String> outputLine = meta.getDataLines().get( data.linesWritten );
 
     for ( int i = 0; i < data.outputRowMeta.size(); i++ ) {
-      if ( meta.isSetEmptyString()[i] ) {
+      if ( meta.isSetEmptyString()[ i ] ) {
         // Set empty string
-        outputRowData[i] = StringUtil.EMPTY_STRING;
+        outputRowData[ i ] = StringUtil.EMPTY_STRING;
       } else {
 
         ValueMetaInterface valueMeta = data.outputRowMeta.getValueMeta( i );
         ValueMetaInterface convertMeta = data.convertMeta.getValueMeta( i );
         String valueData = outputLine.get( i );
 
-        outputRowData[i] = valueMeta.convertDataFromString( valueData, convertMeta, null, null, 0 );
+        if ( valueData != null && valueMeta.isNull( valueData ) ) {
```

Comment:
It is meaningless to invoke isNull(null) if you already know the object is null.  In this case, you know because you've duplicated some of the logic in the isNull() function before you call it.

This type of comparison is usually intended to safety check a reference being passed into a function that might not properly handle a null being passed in.

However, ValueMetaBase.isNull() already does null checking inside of it since that is the whole purpose of the function, and it returns a boolean value.  If you pass a null into that function, it will return true on line 3318, or 3309 if the field is a binary storage field.

Doing the valueData != null comparison as a safety check is unnecessary, and it is redundant to the code that already exists in isNull().

That said, it is harmless, and I'd rather see the PR just merged rather than force a change unless there was already other revisions that needed to be made.